nat-lab/tests/test_mesh_connected_socket.py

```diff
@@ -0,0 +1,95 @@
+import asyncio
+import pytest
+import telio
+from contextlib import AsyncExitStack
+from helpers import SetupParameters, setup_mesh_nodes
+from telio import PathType, State
+from telio_features import TelioFeatures, Direct
+from utils.connection_util import ConnectionTag
+from utils.iperf3 import IperfServer, IperfClient, Protocol
+from utils.ping import Ping
+
+
+@pytest.mark.asyncio
+@pytest.mark.long
+@pytest.mark.parametrize(
+    "disable_connected_socket_a",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "disable_connected_socket_b",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "beta_tag",
+    [ConnectionTag.MAC_VM, ConnectionTag.DOCKER_OPEN_INTERNET_CLIENT_2],
+)
+async def test_mesh_connected_socket(
+    disable_connected_socket_a, disable_connected_socket_b, beta_tag
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        features_alpha = TelioFeatures(
+            direct=Direct(providers=["stun", "local"]),
```

Comment:
Here it probably make sense to leave `Direct()` and avoid overriding defaults for `providers`. As the test does not depend on specific providers, it just needs direct connection. Which will be ensured if `direct is not None`.